### PR TITLE
Add clear option

### PR DESCRIPTION
**Pull request by vrunoa**
Hey @sindresorhus I create a PR to integrate the ```clear``` param option to ```first-run```. I've already add a test, [test3.js](https://github.com/Urucas/first-run/blob/master/test3.js), and the option in the README.md

Let me know if this is OK.

### CODE REVIEW

File: index.js [hunk 1]
```diff
@@ -17,6 +17,10 @@ module.exports = function (opts) {
 	}
 
 	var conf = new Configstore('first-run_' + name, {firstRun: true});
+    if ( opts.clear === true ) {
+        conf.set('firstRun', true);
+        return;
+    }
 
 	if (firstRun === undefined) {
```
Comment (sindresorhus) [inline index.js]: This file is tab indented.

File: readme.md [hunk 2]
```diff
@@ -31,7 +31,7 @@ false
 
 ## API
 
-### firstRun([options])
+### firstRun({options})
```
Comment (sindresorhus) [inline readme.md]: `[]` doesn't mean array. It's the convention for signifying an optional argument.

Comment (vrunoa) [inline readme.md]: I misunderstood that. My mistake!

File: index.js (hunk 1 above)
Comment (vrunoa) [inline index.js]: I should remove the tabspaces ? Or set the default like 1 tab = 4 spaces ?

Comment (sindresorhus) [inline index.js]: Use tab indentation, not spaces.

**Comment by sindresorhus** [comment]
Hmm, actually when looking at this having a `clear` option feels a bit weird. I think it would make more sense to have an extra method: `firstRun.clear()`, which would accept the same options as `firstRun`. What do you think?

**Comment by vrunoa** [comment]
Yeah ```firstRun.clear()``` is definitely more intuitive. Let me update the code and update this PR. But tomorrow, 'cause it's 5am here. I should go to sleep. :sleepy:

**Comment by vrunoa** [comment]
What do you think of this implementation @sindresorhus  ? [https://github.com/Urucas/first-run/blob/master/index.js](https://github.com/Urucas/first-run/blob/master/index.js)

**Comment by sindresorhus** [comment]
I'd prefer to keep it simple. Using a prototype gives us no benefit here. Just add a `.clear()` method to the `module.exports`. Having an internal `getConf` makes sense though, just not the prototype stuff.

**Comment by vrunoa** [comment]
Yeah, makes more sense to just add ```.clear``` to the ```module.exports```.

File: index.js [hunk 3]
```diff
@@ -2,29 +2,35 @@
...
 
-	if (!name) {
-		delete require.cache[__filename];
-		name = require(lookup('package.json', {
-			cwd: path.dirname(module.parent.filename)
-		})).name;
-	}
+function clear(opts) {
+  var conf = getConf(opts);
+  conf.set('firstRun', true);
```
Comment (sindresorhus) [inline index.js]:
```js
getConf(opts).set('firstRun', true);
```

File: index.js [hunk 4]
```diff
@@ -2,29 +2,35 @@
...
-	}
-
-	if (firstRun === true) {
-		conf.set('firstRun', false);
-	}
-
-	return firstRun;
-};
+module.exports = firstRun;
+module.exports["clear"] = clear;
```
Comment (sindresorhus) [inline index.js]:
```js
module.exports.clear = clear;
``

**Comment by sindresorhus** [comment]
Also, don't change the indentaiton. It's tab-indented.

**Comment by vrunoa** [comment]
It was tab-indented, but with 2 spaces. I believe now it's properly tab-indented.

**Comment by arthurvr** [comment]
> It was tab-indented, but with 2 spaces.

How can something be tab-indented with spaces? :p 

Right now the code uses 4 spaces, we use tabs. We use an [EditorConfig](http://editorconfig.org/) file to prevent issues like this, by the way.

**Comment by vrunoa** [comment]
Oh cool, didn't know about EditorConfig, thanks for the info @arthurvr. Just added to vim. I believe now is properly tab-indented, 99% sure.

**Comment by arthurvr** [comment]
@vrunoa Yep, now the indentation is correct :)

**Comment by sindresorhus** [comment]
Thanks. Published as 1.1.0 ;)

**Comment by sindresorhus** [comment]
Thanks. Published as 1.1.0 ;)

**Comment by vrunoa** [comment]
:+1:

**Pull request closed without merging**